src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Fixer.cs

```diff
@@ -76,10 +76,17 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
                 return;
             }
 
+            // Ignore parenthesized operations
+            IOperation? instanceGenericOperation = invocation.Instance;
```

Comment:
I think the codefix does too much. The analyzer can add a properties dictionary to the diagnostic which can be retrieved here.

Reply:
Do you have an example? I don't know how to do what you're suggesting.

Reply:
I _think_ you're referring to the fact that I can tell the analyzer where to put squiggles , but I can also indicate what needs to be changed by the fixer, so that the fixer doesn't need to do the whole search again. Correct?

Reply:
Example, in the analyzer:

https://github.com/dotnet/roslyn-analyzers/blob/2b6ab8d727ce73a78bcbf026ac75ea8a7c804daf/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArrays.cs#L139-L142

In the codefix:

https://github.com/dotnet/roslyn-analyzers/blob/2b6ab8d727ce73a78bcbf026ac75ea8a7c804daf/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArrays.Fixer.cs#L42

Basically you just add "more info" to the diagnostic as a `Dictionary<string, string>` which is retrievable in the codefix. This avoids the need to re-calculate things that were already calculated in the analyzer.

Reply:
Thanks.

Things I think I can remove from the fixer because it might not be necessary to verify them:

- These checks that are probably unnecessary: https://github.com/dotnet/roslyn-analyzers/blob/59c0115ec26fb64e0db8f2883a267dbe312bbee6/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Fixer.cs#L58-L73
- This debug assert: https://github.com/dotnet/roslyn-analyzers/blob/59c0115ec26fb64e0db8f2883a267dbe312bbee6/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Fixer.cs#L107

Things I can't remove:

- This isn't a string: https://github.com/dotnet/roslyn-analyzers/blob/59c0115ec26fb64e0db8f2883a267dbe312bbee6/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Fixer.cs#L50-L54
- The parenthesis operand digging: https://github.com/dotnet/roslyn-analyzers/blob/59c0115ec26fb64e0db8f2883a267dbe312bbee6/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Fixer.cs#L76-L80
- The instanceOperation and the caseChangingMethodName, they are both passed down to the fixer method and the second one isn't even calculated in the analyzer: https://github.com/dotnet/roslyn-analyzers/blob/59c0115ec26fb64e0db8f2883a267dbe312bbee6/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Fixer.cs#L85-L101

Reply:
And I couldn't find anything to send as properties because it's not useful as a string.

Reply:
The one you mentioned at the top is an instance object. Were you suggesting that I pass it somehow in the properties dictionary?

Reply:
Yeah properties can only be strings.
I missed that you need the actual operation, I was thinking only instanceOperation.TargetMethod.Name is relevant. You can keep as it's now.